app/utils/antivirus.py

```diff
@@ -0,0 +1,55 @@
+import requests
+from flask import current_app
+
+
+class AntivirusError(Exception):
+    def __init__(self, message=None, status_code=None):
+        self.message = message
+        self.status_code = status_code
+
+    @classmethod
+    def from_exception(cls, e):
+        try:
+            message = e.response.json()['error']
+            status_code = e.response.status_code
+        except (TypeError, ValueError, AttributeError, KeyError):
+            message = 'connection error'
+            status_code = 503
+
+        return cls(message, status_code)
+
+
+class AntivirusClient:
+    def __init__(self, api_host=None, auth_token=None):
+        self.api_host = api_host
+        self.auth_token = auth_token
+
+    def init_app(self, app):
+        self.api_host = app.config['ANTIVIRUS_API_HOST']
+        self.auth_token = app.config['ANTIVIRUS_API_KEY']
+
+    def scan(self, document):
```

Comment:
Might be worth a comment on the format of document or change the base64_document_string or something

Reply:
It's not base64, it's a direct IO stream

Reply:
Hence the need for the comment :)

Reply:
It turns out I can pass the original request FileStorage object since it proxies `.seek`, so I've changed the call in the upload view and this interface now matches how `document` argument is used in the DocumentStore.

I'll rename them both to e.g. `document_stream`, but it's a bit weird because both could use a string just as well apart from the `.seek(0)` used to reset the stream in AntivirusClient (which could be wrapped in a condition but then it'd have to recognize both `io.*` classes as well as werkzeug's FileStorage).